lua/entities/kleiner/sv_spawning.lua

```diff
@@ -0,0 +1,69 @@
+﻿-- This file is subject to copyright - contact swampservers@gmail.com for more information.
+-- INSTALL: CINEMA
+CreateConVar("kleiner_spawncount", "18", FCVAR_ARCHIVE + FCVAR_PROTECTED, "Number of Kleiner NPCs to spawn automatically.", 0, 400)
+KLEINER_NPCS = KLEINER_NPCS or {}
+KLEINER_NPCS_FILTER = KLEINER_NPCS_FILTER or {}
+KLEINER_NPC_TARGETS = KLEINER_NPC_TARGETS or {}
+KLEINER_NPCS_CURRENT_NUMBER = KLEINER_NPCS_CURRENT_NUMBER or 0
+
+timer.Create("kleiner_spawner", 0.5, 0, function()
+    if (table.Count(KLEINER_NPCS) < (GetConVar("kleiner_spawncount"):GetInt() or 0)) then
+        local newkleiner = ents.Create("kleiner")
+
+        if (IsValid(newkleiner)) then
+            local spawnpoint = gmod.GetGamemode().SpawnPoints and table.Random(gmod.GetGamemode().SpawnPoints):GetPos() or Vector(0, 0, 16)
+            newkleiner:SetPos(spawnpoint)
+            newkleiner:Spawn()
+            newkleiner:Activate()
+        end
+    end
+end)
+
+hook.Add("EntityRemoved", "kleiner_npc_kleanup", function(ent)
+    if (ent:GetClass() == "kleiner" or ent:IsPlayer()) then
+        KLEINER_NPCS[ent] = nil
+        KLEINER_NPC_TARGETS[ent] = nil
+        KLEINER_NPCS_CURRENT_NUMBER = table.Count(KLEINER_NPCS)
+    end
+end)
+
+hook.Add("OnEntityCreated", "kleiner_npc_register", function(ent)
```

Comment:
```
if ent:GetClass() == "kleiner" then
        KLEINER_NPCS[ent] = true
        KLEINER_NPCS_CURRENT_NUMBER = table.Count(KLEINER_NPCS)
elseif ent:IsPlayer() then
        KLEINER_NPC_TARGETS[ent] = true
end
```

Reply:
yeah, i think the pathfinding budget is divided across the number of kleiners